sycl/test-e2e/ProfilingTag/common.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <sycl/sycl.hpp>
+
+#define CHECK(Counter, Check)                                                  \
+  if (!(Check)) {                                                              \
+    std::cout << #Check << " Failed!" << std::endl;                            \
+    ++Counter;                                                                 \
+  }
+
+int run_test_on_queue(sycl::queue &Queue) {
+  Queue.single_task([]() {
+    volatile float X = 1.0f;
+    for (size_t I = 0; I < 200; ++I)
+      X = sycl::sin(X);
+  });
+  sycl::event StartTagE =
+      sycl::ext::oneapi::experimental::submit_profiling_tag(Queue);
+  sycl::event E1 = Queue.single_task([]() {
+    volatile float X = 1.0f;
+    for (size_t I = 0; I < 200; ++I)
+      X = sycl::sin(X);
+  });
+  sycl::event E2 = Queue.single_task([]() {
+    volatile float X = 1.0f;
+    for (size_t I = 0; I < 200; ++I)
+      X = sycl::cos(X);
+  });
+  sycl::event EndTagE =
+      sycl::ext::oneapi::experimental::submit_profiling_tag(Queue);
+
+  Queue.wait();
+
+  int Failures = 0;
+
+  uint64_t StartTagSubmit =
+      StartTagE
+          .get_profiling_info<sycl::info::event_profiling::command_submit>();
+  uint64_t StartTagEnd =
+      StartTagE.get_profiling_info<sycl::info::event_profiling::command_end>();
+  uint64_t EndTagSubmit =
+      EndTagE.get_profiling_info<sycl::info::event_profiling::command_submit>();
+  uint64_t EndTagEnd =
+      EndTagE.get_profiling_info<sycl::info::event_profiling::command_end>();
+
+  std::cout << "StartTagSubmit: " << StartTagSubmit << std::endl;
+  std::cout << "StartTagEnd: " << StartTagEnd << std::endl;
+  std::cout << "EndTagSubmit: " << EndTagSubmit << std::endl;
+  std::cout << "EndTagEnd: " << EndTagEnd << std::endl;
+
+  CHECK(Failures, StartTagSubmit != 0)
+  CHECK(Failures, StartTagEnd != 0)
+  CHECK(Failures, EndTagSubmit != 0)
+  CHECK(Failures, StartTagSubmit != 0)
+
+  CHECK(Failures, StartTagSubmit <= StartTagEnd)
+  CHECK(Failures, EndTagSubmit <= EndTagEnd)
+  CHECK(Failures, StartTagEnd <= EndTagEnd)
+
+  if (Queue.has_property<sycl::property::queue::enable_profiling>()) {
+    uint64_t E1Start =
+        E1.get_profiling_info<sycl::info::event_profiling::command_start>();
+    uint64_t E1End =
+        E1.get_profiling_info<sycl::info::event_profiling::command_end>();
+    uint64_t E2Start =
+        E2.get_profiling_info<sycl::info::event_profiling::command_start>();
+    uint64_t E2End =
+        E2.get_profiling_info<sycl::info::event_profiling::command_end>();
```

Comment:
Likewise, why don't you check `command_submit` here?

Reply:
The intention here is to make sure the commands start execution after the tags have executed. I can't think of anything we can check w.r.t. submit timing that is necessarily relevant to the profiling tags we have submitted.